src/halmos/utils.py

```diff
@@ -182,8 +182,13 @@ def extract_string_argument(calldata: BitVecRef, arg_idx: int):
     return string_bytes.decode("utf-8") if is_concrete(string_bytes) else string_bytes
 
 
-def extract_bytes(data: BitVecRef, byte_offset: int, size_bytes: int) -> BitVecRef:
+def extract_bytes(
+    data: Optional[BitVecRef], byte_offset: int, size_bytes: int
+) -> BitVecRef:
     """Extract bytes from calldata. Zero-pad if out of bounds."""
+    if data is None:
+        return BitVecVal(0, size_bytes * 8)
+
```

Comment:
could you add a regression test that would fail without this fix? (it's not clear when None can be passed to this function.) also, does this fix the `vm.etch(x, "")` case as well?

Reply:
it's related but separate from the `vm.etch(x, "")` issue, we don't have a very nice way to deal with empty bytes in general. They get sometimes represented as `bytes()`, sometimes as `[]`, sometimes as None

the test that revealed this issue is `check_ecrecover_invalidCallReturnsNothing`, in the following line:
`let succ := call(gas(), ECRECOVER_PRECOMPILE, 0, 0, 0, 0, 0)`

Reply:
thanks. now i can see where None comes from. as you mentioned, i agree that we need a better way to handle nicely empty bytes. could you please open an issue about this to discuss further later?